### DISCUSSION TITLE
bevy_render: `out of memory error` and `WebGL: INVALID_ENUM: framebufferTexture2D: invalid attachment` on mobile web safari

### DISCUSSION THREAD

**Discussion by v-kat** [bevyengine/bevy#12678] [Q&A]
## Bevy version

0.13.1

```ignore
AdapterInfo { name: "Apple GPU", vendor: 4203, device: 0, device_type: Other, driver: "", driver_info: "", backend: Gl }
```

## What you did

I have a simple web game with a login page using `webgl2, bevy_render, and webp` when I try to open it on iOS I was seeing a `WebGL: INVALID_ENUM: framebufferTexture2D: invalid attachment` error with a `release` build and a 
```ERROR /home/name/.cargo/git/checkouts/bevy-ead6b24628909bd0/13fbf52/crates/bevy_asset/src/server/mod.rs:302 Failed to load asset 'image.webp' with asset loader 'bevy_render::texture::image_loader::ImageLoader': Could not load texture file: Error reading image file image.webp: failed to load an image: out of memory, this is an error in `bevy_render`.``` error with a `dev` build. The asset is only 363Kb.

## What went wrong

The game isn't playable on mobile safari

**Comment by alice-i-cecile** [comment]
I've converted this to a discussion for now, as it's not clear how universal this is, or how to reproduce / fix it.

How are you doing asset loading? Can you share your project in a runnable state, ideally with source? Can you make a minimal reproduction?

**Comment by v-kat** [comment]
https://github.com/v-kat/bevy_bad_render_bug_repo

It's a simple webp bevy_sprite reproduction of `WebGL: INVALID_ENUM: framebufferTexture2D: invalid attachment` on mobile web safari. 

![error](https://github.com/bevyengine/bevy/assets/1297926/5366d109-1b16-4581-9fcc-0d74d95d2a03)
![expanded](https://github.com/bevyengine/bevy/assets/1297926/4cbfbebe-4142-4b5c-acb2-9b41f923a2e4)

**Reply by v-kat** [reply]
Actually this might only be an issue on iOS 15.5. I couldn't repo it on the latest version.

**Comment by v-kat** [comment]
I think the `out of memory error ` is maybe related to https://github.com/WebAssembly/design/issues/1397 or more likely https://github.com/bevyengine/bevy/pull/11399 my total size for assets was less than 16 MB but I might stop trying to batch load everything too (I might try to do an easier exp for that later).

**Reply by v-kat** [reply]
@alice-i-cecile I managed to reproduce another error which also might be related to OOM or something similar by loading 10 larger images quickly with `asset_server.load(format!("tester{}.webp", x));` on iOS. I updated https://github.com/v-kat/bevy_bad_render_bug_repo
![maybe_oom](https://github.com/bevyengine/bevy/assets/1297926/81be22d5-7943-4ea4-a5b8-b9901db5d7a3)
LMK if this is enough for a not very specific render bug. I can try to load my assets more intelligently and will give that a shot in my main project but this also seems troubling....

**Reply by alice-i-cecile** [reply]
Yeah, this smells like the challenge we want to work around in https://github.com/bevyengine/bevy/issues/12653. WASM's memory patterns are quite frustrating, and I'm not sure the best path forward.

**Reply by v-kat** [reply]
I'm doing a bunch of separate asset loading and still can pretty consistently run into the problem. I don't really see any hooks into the engine I can use to change much but maybe I'm missing something. This seems like it's kind of a bug as opposed to a usage issue. (not just missing WASM GC)

**Reply by v-kat** [reply]
@alice-i-cecile should create an issue for https://github.com/bevyengine/bevy/discussions/12678#discussioncomment-8943924 ?

It still seemed very low memory to trigger even with the lack of iOS wasm gc.

**Reply by alice-i-cecile** [reply]
Yes please: this seems surprising to me and it would be good to collect information about these limitations in an issue, even if we can't get a true fix.
